### PR TITLE
Refactor to a concurrent solution

### PR DESCRIPTION
**Pull request by mantzas**
There is a problem with this test since the writing and reading from the config that is passed in could create a data race.
The thing is that only one writes and all the other read which triggers the race detector.
We have 2 solutions for this:

- Remove the -race flag for the tests 
- Implement some wrapper in order to allow serialized access to the field.

### CODE REVIEW

**Comment by mantzas** [comment]
@c0nstantx @Lazaros-Tsigkakos added missing integration tests

File: README.md [hunk 1]
```diff
@@ -27,6 +27,13 @@ The above defines the following fields:
 - Age, which will be seeded with the value `18`, and if exists, overridden with whatever value the env var `ENV_AGE` holds
 - IsAdmin, which will be seeded with the value `true`, and if exists, overridden with whatever value the env var `ENV_AGE` holds and then from consul if the consul seeder and/or watcher are provided.
 
+The fields have to be on of the types that the sync package supports in order to allow concurrent read and write to the fields. The following types are supported:
```
Comment (Lazaros-Tsigkakos) [inline README.md]: `have to be on of the types` should be converted to `have to be one of the types`. Missed an `e`.

File: config/config.go [hunk 2]
```diff
@@ -22,105 +21,142 @@ const (
...
-}
-
-// New creates a new monitor.
-func New(cfg interface{}) (*Config, error) {
-	if cfg == nil {
-		return nil, errors.New("configuration is nil")
+// NewField constructor.
+func NewField(fld *reflect.StructField, val *reflect.Value) (*Field, error) {
+	if !isTypeSupported(fld.Type) {
+		return nil, fmt.Errorf("field %s is not supported(only types from the sync package of harvester)", fld.Name)
```
Comment (Lazaros-Tsigkakos) [inline config/config.go]: Missed an extra space after `supported`.

File: config/config.go [hunk 3]
```diff
@@ -22,105 +21,142 @@ const (
...
-	v.Lock()
-	defer v.Unlock()
-	f := v.cfg.FieldByName(name)
-	switch kind {
-	case reflect.Bool:
-		b, err := strconv.ParseBool(value)
+// Set the value of the field.
+func (f *Field) Set(value string, version uint64) error {
+	if version != 0 && version <= f.version {
+		return fmt.Errorf("version %d is older or same as field's %s version %d", version, f.name, f.version)
```
Comment (Lazaros-Tsigkakos) [inline config/config.go]: You can also try this one if you think that it sounds better: `version %d is older or the same as the field's`.

File: sync/sync.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
+package sync
+
+import (
+	"sync"
+)
+
+// Bool type with concurrent access support.
+type Bool struct {
+	mu    sync.RWMutex
```
Comment (Lazaros-Tsigkakos) [inline sync/sync.go]: You can also use the namining convension `rw` for RWMutex if it sounds better to you.
https://golang.org/pkg/sync/#RWMutex

**Pull request merged**